.dev/docker/env/Dockerfile

```diff
@@ -36,44 +34,56 @@ RUN apt-get update \
       libusb-1.0-0-dev \
       libvtk${VTK_VERSION}-dev \
       libvtk${VTK_VERSION}-qt-dev \
+      lsb-release \
       qtbase5-dev \
       software-properties-common \
+      wget \
+      xvfb \
+ && if [ "$USE_LATEST_CMAKE" = true ] ; then \
+      wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null ; \
+      if wget -q --method=HEAD "https://apt.kitware.com/ubuntu/dists/$(lsb_release -cs)/Release"; then \
+        apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main" ; \
+      else \
+        apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main' ; \
+      fi \
+    fi \
+ && apt-get -V install -y cmake \
+ && if [ "$(lsb_release -sr)" = "18.04" ]; then \
+       apt-get -V install -y nvidia-cuda-toolkit g++-6 ; \
+     else \
+       apt-get -V install -y nvidia-cuda-toolkit-gcc ; \
+     fi \
  && rm -rf /var/lib/apt/lists/*
 
-# Eigen patch (https://eigen.tuxfamily.org/bz/show_bug.cgi?id=1462) to fix issue metioned
-# in https://github.com/PointCloudLibrary/pcl/issues/3729 is available in Eigen 3.3.7
-# Not needed from 20.04 since it is the default version from apt
-RUN if [ `pkg-config --modversion eigen3 | cut -d. -f3` -lt 7 ]; then \
-     wget -qO- https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.gz | tar xz \
-     && apt install -y libblas-dev \
-     && cd eigen-3.3.7 \
+# Environment variable selected by CMake 3.13 and above to select the CUDA compiler.
+ENV CUDAHOSTCXX=/usr/lib/nvidia-cuda-toolkit/bin/g++
+
+# Use libeigen3-dev if it meets the minimal version.
+# In most cases libeigen3-dev is already installed as a dependency of libvtk6-dev & libvtk7-dev, but not in every case (libvtk9 doesn't seem to have this dependency),
+# so install it from apt if the version is sufficient.
+RUN if dpkg --compare-versions $(apt-cache show --no-all-versions libeigen3-dev | grep -P -o 'Version:\s*\K.*') ge ${EIGEN_MINIMUM_VERSION}; then \
+     apt-get -V install -y libeigen3-dev ; \
+    else \
+     wget -qO- https://gitlab.com/libeigen/eigen/-/archive/${EIGEN_MINIMUM_VERSION}/eigen-${EIGEN_MINIMUM_VERSION}.tar.gz | tar xz \
+     && cd eigen-${EIGEN_MINIMUM_VERSION} \
      && mkdir build \
      && cd build \
      && cmake .. \
-     && make install \
+     && make -j$(nproc) install \
```

Comment:
Just here? What about L82

Reply:
Removed whole L82 as usually there is no need to call `make` before `make install` and the call to `make install` already included `-j$(nproc)`